diff --git a/src/Compiler.php b/src/Compiler.php
index 1111111..2222222 100644
--- a/src/Compiler.php
+++ b/src/Compiler.php
@@ -5052,7 +5052,7 @@ class Compiler
      *
      * @return array
      */
-    protected function multiplyMedia(Environment $env = null, $childQueries = null)
+    protected function multiplyMedia(?Environment $env = null, $childQueries = null)
     {
         if (
             ! isset($env) ||
@@ -5144,7 +5144,7 @@ class Compiler
      *
      * @return Environment
      */
-    protected function pushEnv(Block $block = null)
+    protected function pushEnv(?Block $block = null)
     {
         $env = new Environment();
 
@@ -5208,7 +5208,7 @@ class Compiler
      * @param boolean     $shadow
      * @param Environment $env
      */
-    protected function set($name, $value, $shadow = false, Environment $env = null, $valueUnreduced = null)
+    protected function set($name, $value, $shadow = false, ?Environment $env = null, $valueUnreduced = null)
     {
         if (! isset($env)) {
             $env = $this->getStoreEnv();
@@ -5314,7 +5314,7 @@ class Compiler
      *
      * @return mixed
      */
-    public function get($name, $shouldThrow = true, Environment $env = null, $unreduced = false)
+    public function get($name, $shouldThrow = true, ?Environment $env = null, $unreduced = false)
     {
         $normalizedName = $this->normalizeName($name);
         $specialContentKey = self::$namespaces['special'] . $normalizedName;
@@ -5379,7 +5379,7 @@ class Compiler
      *
      * @return bool
      */
-    protected function has($name, Environment $env = null)
+    protected function has($name, ?Environment $env = null)
     {
         return $this->get($name, false, $env) !== null;
     }
diff --git a/src/Parser.php b/src/Parser.php
index 1111111..2222222 100644
--- a/src/Parser.php
+++ b/src/Parser.php
@@ -140,7 +140,7 @@ class Parser
      * @param boolean              $cssOnly
      * @param LoggerInterface|null $logger
      */
-    public function __construct($sourceName, $sourceIndex = 0, $encoding = 'utf-8', Cache $cache = null, $cssOnly = false, LoggerInterface $logger = null)
+    public function __construct($sourceName, $sourceIndex = 0, $encoding = 'utf-8', ?Cache $cache = null, $cssOnly = false, ?LoggerInterface $logger = null)
     {
         $this->sourceName       = $sourceName;
         $this->sourceIndex      = $sourceIndex;
diff --git a/src/Node/Number.php b/src/Node/Number.php
index 1111111..2222222 100644
--- a/src/Node/Number.php
+++ b/src/Node/Number.php
@@ -578,7 +578,7 @@ class Number extends Node implements \ArrayAccess
      *
      * @return string
      */
-    public function output(Compiler $compiler = null)
+    public function output(?Compiler $compiler = null)
     {
         $dimension = round($this->dimension, static::PRECISION);
 
diff --git a/src/Formatter.php b/src/Formatter.php
index 1111111..2222222 100644
--- a/src/Formatter.php
+++ b/src/Formatter.php
@@ -272,7 +272,7 @@ abstract class Formatter
      *
      * @return string
      */
-    public function format(OutputBlock $block, SourceMapGenerator $sourceMapGenerator = null)
+    public function format(OutputBlock $block, ?SourceMapGenerator $sourceMapGenerator = null)
     {
         $this->sourceMapGenerator = null;
 
diff --git a/src/Warn.php b/src/Warn.php
index 1111111..2222222 100644
--- a/src/Warn.php
+++ b/src/Warn.php
@@ -59,7 +59,7 @@ class Warn
      *
      * @return void
      */
-    public static function setCallback(callable $callback = null)
+    public static function setCallback(?callable $callback = null)
     {
         self::$callback = $callback;
     }
